tc/sgx/common/crypto/verify_certificate.cpp

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+#include <openssl/x509v3.h>
+#include <openssl/pem.h>
+
+#include "verify_certificate.h"
+#include "ias-certificates.h"
+#include "c11_support.h"
+
+verify_status_t verify_certificate_chain(const char* cert_pem)
+#ifndef IAS_CA_CERT_REQUIRED
+{
+    return VERIFY_FAILURE; //fail (conservative approach for simulator-mode and in absence of CA certificate)
+}
+#else //IAS_CA_CERT_REQUIRED is defined
+{
+
+    assert(cert_pem!=NULL);
+
+    /* Using the IAS CA certificate as a root of trust. */
+    /* Checking that cert is signed by CA. */
+
+    X509* cacrt;
+    X509* crt;
+
+    BIO* crt_bio = BIO_new_mem_buf((void*)cert_pem, -1);
+    crt = PEM_read_bio_X509(crt_bio, NULL, 0, NULL);
+    assert(crt != NULL);
+
+    BIO* cacrt_bio = BIO_new_mem_buf((void*)ias_report_signing_ca_cert_pem, -1);
```

Comment:
ias_report_signing_ca_cert_pem should be passed to this function as parameter. Remove any references to variables ias_xx in crypto module to keep it generic.

Reply:
Done